pkg/plugins/mellanox/mellanox_plugin.go

```diff
@@ -225,9 +212,7 @@ func mlnxNicFromMap(mstData map[string]string, singlePort, firstPort bool) (*mln
 		return nil, err
 	}
 	fwData.totalVfs = i
-	if fwData.singlePort {
-		linkType = mstData["LINK_TYPE"]
-	} else if fwData.firstPort {
+	if fwData.firstPort {
 		linkType = mstData["LINK_TYPE_P1"]
```

Comment:
What will happen if LINK_TYPE_P1 is not in the map ?
in case the link type is not configurable, the map will not contain LINK_TYPE_P1 norLINK_TYPE_P2 entries.

Reply:
A warning will appear, check line 226